lang/image.bzl

```diff
@@ -211,21 +212,24 @@ def _app_layer_impl(ctx, runfiles = None, emptyfiles = None):
         _binary_name(ctx): _final_file_path(ctx, ctx.executable.binary),
         # Create a directory symlink from <workspace>/external to the runfiles
         # root, since they may be accessed via either path.
-        _external_dir(ctx): _runfiles_dir(ctx),
+        _external_dir(ctx): runfiles_dir(ctx),
     })
 
+    args = [ctx.expand_location(arg, ctx.attr.data) for arg in ctx.attr.args]
```

Comment:
add doc comment above explaining you are expanding locations of the form $(location :some_target)